tests/sentry/middleware/test_ratelimit_middleware.py

```diff
@@ -215,3 +217,46 @@ class TestEndpoint:
             pass
 
         assert get_rate_limit_value("GET", TestEndpoint, "ip") is None
+
+
+class RateLimitHeaderTestEndpoint(Endpoint):
+    permission_classes = (AllowAny,)
+
+    enforce_rate_limit = True
+    rate_limits = {"GET": {RateLimitCategory.IP: RateLimit(1, 100)}}
+
+    def get(self, request):
+        return Response({"ok": True})
+
+
+urlpatterns = [
+    url(r"^/ratelimit$", RateLimitHeaderTestEndpoint.as_view(), name="ratelimit-header-endpoint")
+]
+
+
+@override_settings(ROOT_URLCONF="tests.sentry.middleware.test_ratelimit_middleware")
+class TestRatelimitHeader(APITestCase):
+
+    endpoint = "ratelimit-header-endpoint"
+
+    def test_header_counts(self):
+        """Ensure that the header counts increase properly"""
+        with freeze_time("2000-01-01"):
+            response = self.get_success_response()
+            assert int(response["X-Sentry-Ratelimit-Count"]) == 1
+            assert int(response["X-Sentry-Ratelimit-Max"]) == 1
+            assert int(response["X-Sentry-Ratelimit-Window"]) == 100
+
+            response = self.get_error_response()
+            assert int(response["X-Sentry-Ratelimit-Count"]) == 2
+            assert int(response["X-Sentry-Ratelimit-Max"]) == 1
+            assert int(response["X-Sentry-Ratelimit-Window"]) == 100
+
+    @patch("sentry.ratelimits.utils.can_be_ratelimited")
+    def test_omit_header(self, can_be_ratelimited_patch):
+        """Ensure that functions that can't be rate limited don't have rate limit headers"""
+        can_be_ratelimited_patch.return_value = False
+        response = self.get_response()
+        assert "X-Sentry-Ratelimit-Count" not in response._headers
+        assert "X-Sentry-Ratelimit-Max" not in response._headers
+        assert "X-Sentry-Ratelimit-Window" not in response._headers
```

Comment:
Hey @AniketDas-Tekky - thanks for including me in the review - I have nothing to add but I do wonder if "Max" and "Window" are the best header names for our non-English speaking users. 

Okta uses "Limit" and "Reset" respectively. https://developer.okta.com/docs/reference/rl-best-practices/#check-your-rate-limits-with-okta-s-rate-limit-headers